### PR TITLE
Add better NewKeyCertBundleRotator for easier initialization of keycertbundlerotator.

### PR DESCRIPTION
**Pull request by myidpt**

### CODE REVIEW

File: security/pkg/caclient/rotatorfactory.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	}
+
+	pc, err := platform.NewClient(cfg.Env, cfg.RootCertFile, cfg.KeyFile, cfg.CertChainFile, cfg.CAAddress)
+	if err != nil {
+		return nil, err
+	}
+	cAClient, err := NewCAClient(pc, &grpc.CAGrpcClientImpl{}, cfg.CAAddress, cfg.Org, cfg.RSAKeySize,
+		cfg.RequestedCertTTL, cfg.ForCA, cfg.CSRMaxRetries, cfg.CSRInitialRetrialInterval)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to initialize CAClient: %v", err)
```
Comment (incfly) [inline security/pkg/caclient/rotatorfactory.go]: s/Failed/failed. Go recommends to start with lower case for error message.
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: security/pkg/caclient/rotatorfactory.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to initialize KeyCertBundle: %v", err)
```
Comment (incfly) [inline security/pkg/caclient/rotatorfactory.go]: Here as well.

File: security/pkg/caclient/rotatorfactory_test.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+			config: &Config{
+				CertChainFile: "../platform/testdata/cert-chain-good.pem",
+				Env:           "unspecified",
+			},
+			expectedErr: "",
+		},
+		"Unsupported env": {
+			config: &Config{
+				CertChainFile: "../platform/testdata/cert-chain-good.pem",
+				Env:           "somethig else",
```
Comment (incfly) [inline security/pkg/caclient/rotatorfactory_test.go]: can we change this to an non-spaced word, e.g. nonexisting-env? It'll be more obvious the string comes from argument, not part of string template itself.

File: security/pkg/caclient/rotatorfactory_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+			expectedErr: "invalid env somethig else specified",
+		},
+	}
+
+	for id, c := range testCases {
+		_, err := NewKeyCertBundleRotator(c.config, &mock.FakeKeyCertBundle{})
+
+		if len(c.expectedErr) > 0 {
+			if err == nil {
+				t.Errorf("%s: Succeeded. Error expected: %v", id, err)
```
Comment (incfly) [inline security/pkg/caclient/rotatorfactory_test.go]: lower case here as well.

File: security/pkg/caclient/rotatorfactory_test.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+			} else if err.Error() != c.expectedErr {
+				t.Errorf("%s: incorrect error message: %s VS %s",
+					id, err.Error(), c.expectedErr)
+			}
+		} else if err != nil {
+			t.Errorf("%s: Unexpected Error: %v", id, err)
```
Comment (incfly) [inline security/pkg/caclient/rotatorfactory_test.go]: and here.

File: security/pkg/caclient/rotatorfactory.go [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+	"fmt"
+
+	"istio.io/istio/security/pkg/caclient/grpc"
+	pkiutil "istio.io/istio/security/pkg/pki/util"
+	"istio.io/istio/security/pkg/platform"
+	"istio.io/istio/security/pkg/util"
+)
+
+// NewKeyCertBundleRotator is constructor for keyCertBundleRotatorImpl based on the provided configuration.
+func NewKeyCertBundleRotator(cfg *Config, bundle pkiutil.KeyCertBundle) (KeyCertBundleRotator, error) {
```
Comment (incfly) [inline security/pkg/caclient/rotatorfactory.go]: Really like that we add more sanity checking and test for this :)

Do we need to put it in separate factory file? The constructor is very related with the Rotator itself & user, putting them in the same file would make file tighter. We can still have TestNewKeyCertBundleRotator rotator_test.go

File: security/pkg/caclient/rotatorfactory.go (hunk 1 above)
Comment (myidpt) [reply security/pkg/caclient/rotatorfactory.go]: Sure.

File: security/pkg/caclient/rotatorfactory_test.go (hunk 3 above)
Comment (myidpt) [reply security/pkg/caclient/rotatorfactory_test.go]: Done.

File: security/pkg/caclient/rotatorfactory.go (hunk 6 above)
Comment (myidpt) [reply security/pkg/caclient/rotatorfactory.go]: Sure. I can put them together :)

**Pull request merged**